invalid memory address monitor.listResourceGroupMonitoringHostnames

Caught this panic in `(*monitor).listResourceGroupMonitoringHostnames` (application insights manager) during an arm template deployment with the following logs:

```
time="2019-04-05T09:10:02Z" level=info msg="applying arm template deployment" func=github.com/openshift/openshift-azure/pkg/fakerp.GetDeployer.func1 file="/go/src/github.com/openshift/openshift-azure/pkg/fakerp/fakerp.go:100"
time="2019-04-05T09:10:08Z" level=info msg="waiting for arm template deployment to complete" func=github.com/openshift/openshift-azure/pkg/fakerp.GetDeployer.func1 file="/go/src/github.com/openshift/openshift-azure/pkg/fakerp/fakerp.go:120"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x910151]

goroutine 1 [running]:
main.(*monitor).listResourceGroupMonitoringHostnames.func2(0xc00058e0c0, 0xc000329ac7, 0x1)
	/go/src/github.com/openshift/openshift-azure/cmd/monitoring/monitoring.go:104 +0x291
github.com/openshift/openshift-azure/vendor/k8s.io/apimachinery/pkg/util/wait.WaitFor(0xc0004a4120, 0xc0002e87c0, 0xc00058e060, 0x9a3f60, 0xc000329b20)
	/go/src/github.com/openshift/openshift-azure/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:311 +0x71
github.com/openshift/openshift-azure/vendor/k8s.io/apimachinery/pkg/util/wait.PollUntil(0x2540be400, 0xc0002e87c0, 0xc00058e060, 0xc000314120, 0x23)
	/go/src/github.com/openshift/openshift-azure/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:290 +0x56
github.com/openshift/openshift-azure/vendor/k8s.io/apimachinery/pkg/util/wait.PollInfinite(0x2540be400, 0xc0002e87c0, 0x0, 0x0)
	/go/src/github.com/openshift/openshift-azure/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:264 +0x8a
github.com/openshift/openshift-azure/vendor/k8s.io/apimachinery/pkg/util/wait.PollImmediateInfinite(0x2540be400, 0xc0002e87c0, 0x1, 0x1)
	/go/src/github.com/openshift/openshift-azure/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:281 +0x70
main.(*monitor).listResourceGroupMonitoringHostnames(0xc00008aae0, 0xb0fe20, 0xc000028070, 0xc000020356, 0x24, 0xc00002427e, 0x21, 0x120, 0xc00028fda0, 0xc00028fdd0, ...)
	/go/src/github.com/openshift/openshift-azure/cmd/monitoring/monitoring.go:96 +0x26b
main.(*monitor).run(0xc00008aae0, 0xb0fe20, 0xc000028070, 0xc00008a9c0, 0x0)
	/go/src/github.com/openshift/openshift-azure/cmd/monitoring/monitoring.go:129 +0xdd
main.main()
	/go/src/github.com/openshift/openshift-azure/cmd/monitoring/monitoring.go:273 +0x345
time="2019-04-05T09:18:54Z" level=info msg="waiting for API server healthz" func="github.com/openshift/openshift-azure/pkg/cluster.(*simpleUpgrader).WaitForHealthzStatusOk" file="/go/src/github.com/openshift/openshift-azure/pkg/cluster/healthcheck.go:34"
```

artifacts: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_openshift-azure/1431/pull-ci-openshift-openshift-azure-master-e2e-azure/3401/build-log.txt

/kind test-flake

--- a/cmd/monitoring/monitoring.go
+++ b/cmd/monitoring/monitoring.go
@@ -101,7 +101,9 @@ func (m *monitor) listResourceGroupMonitoringHostnames(ctx context.Context, subs
 				return false, nil
 			}
 
-			ips = append(ips, *iter.Value().IPAddress)
+			if iter.Value().IPAddress != nil {
+				ips = append(ips, *iter.Value().IPAddress)
+			}
 		}
 		if len(ips) == 3 {
 			hostnames = append(hostnames, ips...)

